crates/relayer/src/chain/axon.rs

```diff
@@ -963,7 +973,67 @@ impl AxonChain {
     }
 }
 
-fn convert_err(err: ContractError<ContractProvider>) -> Error {
+impl AxonChain {
+    fn send_message(&mut self, message: Any) -> Result<IbcEventWithHeight, Error> {
+        let tx_receipt = match message.type_url.as_str() {
+            conn_open_init::TYPE_URL => {
+                let msg: contract::MsgConnectionOpenInit = message.try_into()?;
+                let tx_receipt: eyre::Result<Option<TransactionReceipt>> =
+                    self.rt.block_on(async {
+                        Ok(self
+                            .contract
+                            .connection_open_init(msg.clone())
+                            .send()
+                            .await?
+                            .await?)
+                    });
+                tx_receipt.map_err(convert_err)?
+            }
+            _ => {
+                todo!()
+            }
+        };
+        let tx_receipt = tx_receipt.ok_or(Error::send_tx(String::from("fail to send tx")))?;
```

Comment:
I think the transaction receipt contains the solidity event that we need, what if we extract the event from the receipt?